### DISCUSSION TITLE
Creating template sensor for wind speed at 2m high

### DISCUSSION THREAD

**Discussion by pedromrg** [jeroenterheerdt/HAsmartirrigation#487] [Q&A]
Hi there,

Hope anyone can help me out since I cannot get my temaplete sensor to work, I always get:

ValueError: Template error: float got invalid input 'unknown' when rendering template 'sensors: wind_at_2m: friendly_name: Wind Speed at 2m value_template: {{states('[sensor.gw2000a_wind_speed (WSmeasured)]')|float()*(4.87/log((67.8*[1 (H)])-5.42))}}' but no default was specified

I'm using:

sensors:
  wind_at_2m:
    friendly_name: Wind Speed at 2m
    value_template: {{states('[sensor.gw2000a_wind_speed (WSmeasured)]')|float()*(4.87/log((67.8*[1 (H)])-5.42))}}

Can't find where I'm doing it wrong.

**Comment by ChrisHomeAssistant** [comment]
Hi!

I think the syntax of the value is not completely right. It should rather be something like:
`value_template: {{states('sensor.gw2000a_wind_spped') | float * (transformation) }}`

The transformation part might also show an error at the log(). What exactly should be done in this transformation?

**Reply by ChrisHomeAssistant** [reply]
I imagine you are using the sensor described in the documentation.
Hence, your sensor could look like this:
`value_template: {{states('sensor.gw2000a_wind_speed') | float * (4.87/log((67.8*1)-5.42)) }}`

**Reply by pedromrg** [reply]
thanks a lot!
